### PR TITLE
DEV-799: Funds Flow Selector Logic and Transfer Workflow component styles

### PR DESCRIPTION
**Pull request by ShreyaThapa**
- Funds Flow selector logic implemented
  - States use for each dropdown selector
  - Based on options selected, use if else condition to select the related webhooks data JSON
- Transfer workflow component

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ module.exports = {
     "import/no-unresolved": "off", // typescript
     "jsx-a11y/anchor-is-valid": "off", // next/link
     "no-confusing-arrow": "off", // prettier
+    "nonblock-statement-body-position": "off", // prettier
```
Comment (ShreyaThapa) [inline .eslintrc.js:32]: Rule to ignore eslint check

File: app/components/base/Button.tsx [hunk 2]
```diff
@@ -16,28 +16,6 @@ import {
 import { BOX_SHADOW_6 } from "../shadowDepths";
 import { POPPINS } from "../typography";
 
-type Props = {
```
Comment (ShreyaThapa) [inline app/components/base/Button.tsx:19]: Moved Props declaration to below "styles" to stay consistent with other component structures.

File: app/components/base/Button.tsx [hunk 3]
```diff
@@ -121,13 +100,22 @@ const buttonStyles = css`
   }
   &.hollow-white {
     border: 1px solid transparent;
-    background: #ffffff;
+    background: ${WHITE_PRIMARY};
```
Comment (ShreyaThapa) [inline app/components/base/Button.tsx:103]: Replacing hex with imported variable.

File: app/components/base/Button.tsx [hunk 4]
```diff
@@ -104,6 +82,7 @@ const buttonStyles = css`
   &.hollow-light {
     border: 1px solid ${GREY_4};
     color: ${GREY_9};
+    background: ${WHITE_PRIMARY};
```
Comment (ShreyaThapa) [inline app/components/base/Button.tsx:85]: Changing the background color of variant to match updated styles.

File: app/components/base/Button.tsx [hunk 5]
```diff
@@ -121,13 +100,22 @@ const buttonStyles = css`
...
     border: 1px solid transparent;
-    background: #ffffff;
+    background: ${WHITE_PRIMARY};
     color: #2d2d48;
     :hover,
     :focus {
       border-color: ${GREY_5};
     }
   }
+  &.disabled {
```
Comment (ShreyaThapa) [inline app/components/base/Button.tsx:110]: Adding style for disabled <Button>

File: app/components/base/Button.tsx [hunk 6]
```diff
@@ -139,11 +127,56 @@ const StyledLink = styled.a`
...
+  link,
+  isDisabled,
+  onButtonClick,
+}: Props) => (
   <StyledLink
     href={link.href}
     target={link.external && "_blank"}
-    className={`${variant} ${size}`}
+    className={`${variant} ${size} ${isDisabled ? "disabled" : ""}`}
+    onClick={onButtonClick}
```
Comment (ShreyaThapa) [inline app/components/base/Button.tsx:179]: Applying button.disabled style conditionally. 
Adding onClick handler for button

File: app/components/base/FundsFlowSelector.tsx [hunk 7]
```diff
@@ -108,32 +108,6 @@ const customStyles = {
...
-  { value: "ro", label: "Receive-Only User" },
-];
-
-const receiverDestinationOptions = [
-  { value: "balance", label: "Balance" },
-  { value: "bank", label: "Bank" },
-  { value: "r03", label: "R03-Bank" },
-  { value: "card", label: "Debit Card" },
-];
-
```
Comment (ShreyaThapa) [inline app/components/base/FundsFlowSelector.tsx:136]: Removing options from FundFlowSelector component. They will be passed down as props from the parent component.

File: app/components/base/FundsFlowSelector.tsx [hunk 8]
```diff
@@ -159,7 +133,29 @@ const Option = (props: any) => {
...
+function FundsFlowSelector({
+  senderTypeOptions,
+  senderSourceOptions,
+  receiverTypeOptions,
+  receiverDestinationOptions,
+  setSelectedSender,
+  setSelectedSource,
+  setSelectedReceiver,
+  setSelectedDestination,
+}: Props) {
```
Comment (ShreyaThapa) [inline app/components/base/FundsFlowSelector.tsx:158]: Accepting options and state as props.

File: app/components/base/FundsFlowSelector.tsx [hunk 9]
```diff
@@ -169,13 +165,15 @@ function FundsFlowSelector() {
         options={senderTypeOptions}
         components={{ DropdownIndicator, Option }}
         placeholder="Select Sender Type"
+        onChange={setSelectedSender}
```
Comment (ShreyaThapa) [inline app/components/base/FundsFlowSelector.tsx:168]: Sets state whenever an option is selected.

File: app/components/base/TransferWorkflow.tsx [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+  { value: "vcr", label: "Verified Customer" },
+  // { value: "cr", label: "Unverified Customer" },
+  // { value: "ro", label: "Receive-Only User" },
+];
+
+const receiverDestinationOptions = [
+  // { value: "balance", label: "Balance" },
+  { value: "bank", label: "Bank" },
+  // { value: "r03", label: "R03-Bank" },
+  // { value: "card", label: "Debit Card" },
```
Comment (ShreyaThapa) [inline app/components/base/TransferWorkflow.tsx:80]: Other options are commented out to demonstrate mvp with one working combination.

File: app/components/base/TransferWorkflow.tsx [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+];
+
+// possible Funds Flow combinations
+
+const fundsFlowCombinations = {
+  one: ["vcr", "bank", "vcr", "bank"],
+  // all possible comibnations as arrays, for example,
+  // two: ["vcr", "balance", "vcr", "bank"]
+  // three: ["cr", "bank", "vcr", "bank"]
```
Comment (ShreyaThapa) [inline app/components/base/TransferWorkflow.tsx:89]: Will need to fill out all possible combinations while working on gathering data later on.

File: app/components/base/__tests__/FundFlowSelector.test.tsx [hunk 12]
```diff
@@ -1,8 +1,48 @@
...
+        senderSourceOptions={senderSourceOptions}
+        receiverTypeOptions={receiverTypeOptions}
+        receiverDestinationOptions={receiverDestinationOptions}
+        setSelectedSender={setState}
+        setSelectedSource={setState}
+        setSelectedReceiver={setState}
+        setSelectedDestination={setState}
+      />
+    )
+    .toJSON();
```
Comment (ShreyaThapa) [inline app/components/base/__tests__/FundFlowSelector.test.tsx:45]: Update test with props.

File: app/components/base/__tests__/__snapshots__/Banner.test.tsx.snap [hunk 13]
```diff
@@ -39,7 +39,7 @@ exports[`Banner - with Button 1`] = `
       Step-by-step instructions to get you set up in the Dwolla API
     </div>
     <a
-      className="primary large css-11e3fwu-StyledLink-buttonStyles-buttonStyles ek3ainu1"
+      className="primary large  css-5fh7ca-StyledLink-buttonStyles-buttonStyles ek3ainu1"
```
Comment (ShreyaThapa) [inline app/components/base/__tests__/__snapshots__/Banner.test.tsx.snap:42]: Updated test due to new styles in <Button> component.

File: app/components/base/__tests__/__snapshots__/Button.test.tsx.snap [hunk 14]
```diff
@@ -2,7 +2,7 @@
 
 exports[`Button 1`] = `
 <button
-  className="primary standard css-1kpf29c-StyledButton-buttonStyles-buttonStyles ek3ainu0"
+  className="primary standard  css-4oxaqu-StyledButton-buttonStyles-buttonStyles ek3ainu0"
```
Comment (ShreyaThapa) [inline app/components/base/__tests__/__snapshots__/Button.test.tsx.snap:5]: Updated test due to new styles in component.

File: assets/transfer-workflow-data/vcr-bank-to-vcr-bank.json [hunk 15]
```diff
@@ -0,0 +1,272 @@
...
+            "created": "2019-03-06T03:48:38.469Z",
+            "id": "d97560a7-7d81-4732-916e-7d18c1ec1928",
+            "resourceId": "e0bef963-c23f-e911-8116-917813db6033",
+            "topic": "customer_bank_transfer_completed"
+          }
+        }
+      ]
+    }
+  ]
+}
```
Comment (ShreyaThapa) [inline assets/transfer-workflow-data/vcr-bank-to-vcr-bank.json:272]: Example webhooks JSON data for one combination "vcr-bank-to-vcr-bank"

File: package.json [hunk 16]
```diff
@@ -59,6 +59,7 @@
     "jest": "^25.1.0",
     "jest-transformer-mdx": "^2.1.0",
     "lodash.groupby": "^4.6.0",
+    "lodash.isequal": "^4.5.0",
```
Comment (ShreyaThapa) [inline package.json:62]: Importing lodash.isequal for comparing arrays.

File: stories/base/FundsFlowSelector.stories.tsx [hunk 17]
```diff
@@ -3,6 +3,46 @@ import { storiesOf } from "@storybook/react";
...
+  { value: "ro", label: "Receive-Only User" },
+];
+
+const receiverDestinationOptions = [
+  { value: "balance", label: "Balance" },
+  { value: "bank", label: "Bank" },
+  { value: "r03", label: "R03-Bank" },
+  { value: "card", label: "Debit Card" },
+];
+
```
Comment (ShreyaThapa) [inline stories/base/FundsFlowSelector.stories.tsx:31]: Updating story for FundsFlowSelector with props.

**Review by tarricsookdeo — APPROVED** [review]

**Review by tarricsookdeo — APPROVED** [review]

**Review by tarricsookdeo — APPROVED** [review]

**Review by tarricsookdeo — APPROVED** [review]
Looks good!

**Pull request merged**